Doc/howto/a-conceptual-overview-of-asyncio.rst

```diff
@@ -0,0 +1,480 @@
+.. _a-conceputal-overview-of-asyncio:
+
+********************************
+A Conceptual Overview of asyncio
+********************************
+
+:Author: Alexander Nordin
+
+This article seeks to help you build a sturdy mental model of how asyncio
+fundamentally works.
+Something that will help you understand the how and why behind the recommended
+patterns.
+
+During my own asyncio learning process, a few aspects particually drove my
+curiosity (read: drove me nuts).
+You should be able to comfortably answer all these questions by the end
+of this article.
+
+- What's roughly happening behind the scenes when an object is ``await``\ ed?
+- How does asyncio differentiate between a task which doesn't need CPU-time
+  to make progress towards completion, for example, a network request or file
+  read as opposed to a task that does need CPU-time to make progress, like
+  computing n-factorial?
+- How would I go about writing my own asynchronous variant of some operation?
+  Something like an async sleep, database request, and so on.
+
+--------------------------------------------
+A conceptual overview part 1: the high-level
+--------------------------------------------
+
+In part 1, we'll cover the main, high-level building blocks of asyncio: the
+event loop, coroutine functions, coroutine objects, tasks and ``await``.
+
+==========
+Event Loop
+==========
+
+Everything in asyncio happens relative to the event loop.
+It's the star of the show.
+It's kind of like an orchestra conductor or military general.
+It's behind the scenes managing resources.
+Some power is explicitly granted to it, but a lot of its ability to get things
+done comes from the respect and cooperation of its subordinates.
```

Comment:
```suggestion
Some power is explicitly granted to it, but a lot of its ability to get things
done comes from the respect and cooperation of its users.
```

Reply:
Teammates works well. Good suggestion.

Reply:
Right on. Done 👍

Reply:
I think this is done, so I'm marking as resolved.